src/dgipy/network_graph.py

```diff
@@ -90,162 +108,25 @@ def create_network(
-        nodes_by_group[node_color]["node_size"].append(node_size)
-        nodes_by_group[node_color]["neighbors"].append(list(graph.neighbors(node)))
-
-    trace_nodes = []
-
-    for _, node in nodes_by_group.items():  # noqa: PERF102
-        trace_group = go.Scatter(
-            x=node["node_x"],
-            y=node["node_y"],
-            mode="markers",
-            marker={
-                "symbol": "circle",
-                "size": node["node_size"],
-                "color": node["node_color"],
-            },
-            text=node["node_text"],
-            name=node["legend_name"],
-            customdata=node["neighbors"],
-            hoverinfo="text",
-            visible=True,
-            showlegend=True,
-        )
-        trace_nodes.append(trace_group)
-
-    return trace_nodes
-
-
-def _create_trace_edges(graph: nx.Graph, pos: dict) -> go.Scatter:
-    edge_x = []
-    edge_y = []
-
-    i_edge_x = []
-    i_edge_y = []
-    i_edge_id = []
-
-    for edge in graph.edges():
-        x0, y0 = pos[edge[0]]
-        x1, y1 = pos[edge[1]]
-        edge_x.append(x0)
-        edge_x.append(x1)
-        edge_x.append(None)
-        edge_y.append(y0)
-        edge_y.append(y1)
-        edge_y.append(None)
-
-        i_edge_x.append((x0 + x1) / 2)
-        i_edge_y.append((y0 + y1) / 2)
-        i_edge_id.append(graph.edges[edge]["id"])
-
-    trace_edges = go.Scatter(
-        x=edge_x,
-        y=edge_y,
-        mode="lines",
-        line={"width": 0.5, "color": "gray"},
-        hoverinfo="none",
-        showlegend=False,
-    )
-
-    i_trace_edges = go.Scatter(
-        x=i_edge_x,
-        y=i_edge_y,
-        mode="markers",
-        marker_size=0.5,
-        text=i_edge_id,
-        hoverinfo="text",
-        showlegend=False,
-    )
-
-    return trace_edges, i_trace_edges
-
-
-def generate_json(graph: nx.Graph) -> dict:
-    """Generate a JSON representation of a networkx graph
+def generate_cytoscape(graph: nx.Graph) -> dict:
+    """Create a cytoscape graph representing interactions between genes and drugs
 
-    :param graph: networkx graph to be formatted as a JSON
-    :return: a dictionary representing the JSON data of the graph
+    :param graph: networkx graph to be formatted as a cytoscape graph
+    :return: a cytoscape graph of drug-gene interactions
     """
-    return nx.node_link_data(graph)
+    pos = nx.spring_layout(graph, seed=LAYOUT_SEED, scale=4000)
+    cytoscape_data = nx.cytoscape_data(graph)["elements"]
+    cytoscape_node_data = cytoscape_data["nodes"]
+    cytoscape_edge_data = cytoscape_data["edges"]
+    for node in range(len(cytoscape_node_data)):
+        node_pos = pos[cytoscape_node_data[node]["data"]["id"]]
+        node_pos = {
+            "position": {"x": int(node_pos[0].item()), "y": int(node_pos[1].item())}
+        }
+        cytoscape_node_data[node].update(node_pos)
+    return cytoscape_node_data + cytoscape_edge_data
```

Comment:
this is way more compact that before. I am pro this change to cytoscape.

How much of a dependency burden is there with dash_cytoscape? Iirc we will be modularizing this separately from the dgipy search functionality anyway, but I just want to get a sense of how much the user will have to install beforehand to use this

Reply:
dash_cytoscape doesn't have much of a dependency burden. Besides dash and networkx (to provide the layout of the graph), cytoscape doesn't really have any other dependencies that we'd need a user to install for cytoscape to work.